tensorflow/core/kernels/pooling_ops_3d.cc

```diff
@@ -837,6 +845,698 @@ TF_CALL_float(REGISTER_GPU_KERNELS) TF_CALL_half(REGISTER_GPU_KERNELS)
 
 #endif  // GOOGLE_CUDA
 
+#ifdef TENSORFLOW_USE_SYCL
```

Comment:
This file is starting to get long. I wonder if we should split it. @benoitsteiner WDYT?